main.py

```diff
@@ -22,10 +23,19 @@
 class ApplicationWindow(QtWidgets.QMainWindow):
     """The main GUI window."""
 
+    # Define a new signal, used to add_frames from the seperate thread
+    add_frame = QtCore.pyqtSignal('PyQt_PyObject', 'PyQt_PyObject', 'PyQt_PyObject')
+    # Used when the add stream thread fails
+    fail_add_stream = QtCore.pyqtSignal('PyQt_PyObject', 'PyQt_PyObject')
```

Comment:
When coding this I actually could not get other types to work. However this time around it worked, so fixed!